theano/gof/toolbox.py

```diff
@@ -200,7 +201,27 @@ def on_detach(self, fgraph):
 
     def validate_(self, fgraph):
         t0 = time.time()
-        ret = fgraph.execute_callbacks('validate')
+        try:
```

Comment:
I think it would be simpler to just do the printing here and not at replace_validate. This would make the code much simpler.

But otherwise, it is working correctly.